src/twisted/conch/test/test_endpoints.py

```diff
@@ -1435,31 +1435,37 @@ def test_cleanupConnectionImmediately(self):
         helper.cleanupConnection(object(), True)
 
 
+class _WriteDiscarder(BytesIO):
+    def write(self, data: object, /) -> int:
+        """
+        Discard writes because we are emulating a console object, where they'd
+        go to the screen, not back into the buffer to be read like an a+ file.
+        """
+        return 0
+
+
 class _PTYPath:
     """
     A L{FilePath}-like object which can be opened to create a L{_ReadFile} with
     certain contents.
     """
 
-    def __init__(self, contents):
+    def __init__(self, contents: bytes) -> None:
         """
         @param contents: L{bytes} which will be the contents of the
             L{_ReadFile} this path can open.
         """
         self.contents = contents
 
-    def open(self, mode):
+    def open(self, mode: str) -> IO[bytes]:
         """
-        If the mode is r+, return a L{_ReadFile} with the contents given to
-        this path's initializer.
-
-        @raise OSError: If the mode is unsupported.
+        If the mode is r+, return a file with the given contents as a line.
 
-        @return: A L{_ReadFile} instance
+        @return: a buffer of the given contents, which will discard any writes
+            given to it.
         """
-        if mode == "rb+":
-            return _ReadFile(self.contents)
-        raise OSError(ENOSYS, "Function not implemented")
+        assert mode == "r+"
```

Comment:
should't we be explicit about binary mode?

Reply:
Nope, once again, `FilePath` is always binary mode.